Sources/Starknet/Providers/StarknetProviderProtocol.swift

```diff
@@ -163,7 +163,7 @@ public extension StarknetProviderProtocol {
     /// - Parameters:
     ///  -  transaction: transaction for which the fee should be estimated.
     /// - Returns: Fee estimate
-    func estimateFee(for transaction: any StarknetSequencerTransaction) async throws -> StarknetFeeEstimate {
+    func estimateFee(for transaction: any StarknetTransaction) async throws -> StarknetFeeEstimate {
```

Comment:
Probably not in this PR, but I think we should consider not supporting `simulateTransaction` and `estimateFee` for any transaction.

It is misleading, user can assume any transaction can be estimated. In reality, we shouldn't allow L1 handler tx there. Declare is also tricky: techichally, we can estimate fee for it, but only if the transaction is fetched from the network. Since we don't support declare in `simulateTransaction`, I think we should restrict this (probably unused) case.

Reply:
agreed

Reply:
#128